sys/drv/rtl8139.c

```diff
@@ -32,18 +37,72 @@ static int rtl_reset(rtl8139_state_t *state) {
   return 1;
 }
 
-static int rtl8139_attach(device_t *dev) {
+static intr_filter_t rtl8139_intr(void *data) {
+  rtl8139_state_t *state = data;
+  // TODO: add enqueue routines
+  bus_write_2(state->regs, RL_ISR, RL_ISR_RX_OK);
+  return IF_FILTERED;
+}
+
+static int set_mem_contig_address(rtl8139_state_t *state) {
```

Comment:
We must get rid of it in the future (in some driver infrastructure related PR).

Reply:
This function is so ugly that I insist it should be done sooner than later.

Reply:
@cahirwpz thank you for `kmem_alloc_contig`